### DISCUSSION TITLE
reolink sent me a firmware for rlc-410w with hw version IPC_51316M greater then S07

### DISCUSSION THREAD

**Discussion by clsferguson** [AT0myks/reolink-fw-archive#12] [Firmware sharing]
It allows for definable interframe. I'm hoping they get back to me soon with one for a fixed frame rate as well.

[IPC_51316M.757_21121503.RLC-410W.ov4689.4MP.WIFI1021.REOLINK.zip](https://github.com/AT0myks/reolink-fw-archive/files/11315740/IPC_51316M.757_21121503.RLC-410W.ov4689.4MP.WIFI1021.REOLINK.zip)

This specific firmware does not work with some of the older models that start with S07.
![image](https://user-images.githubusercontent.com/48876201/234154696-e32fc5a1-bb23-4b06-8fc5-e82e67395299.png)

**Comment by AT0myks** [comment]
Thanks, that's interesting. I don't think I've ever seen restrictions on compatibility more specific than hardware version. Is this an image and explanation they sent you along with the firmware, or is it something you can find on their website? I wonder what would happen if you installed it on the older models.

**Reply by clsferguson** [reply]
Here is the context:

I asked:
> just wondering if you can send me the latest firmware for a rlc-410w with hardware IPC_51316M
>  
> I can't find it in the download center
>  
> thanks.

They replied:

> Dear Cory ,
> 
> This is Olivia from Reolink Support Team.
> Thank you for reaching out to us regarding the firmware for 410W.
> 
> As for the RLC-410W (IPC_51316M), the latest version is v3.0.0.389_ 21062202.
> This specific firmware is not posted on the Download Center because it does not work with some of the older models that start with S07.
> 
> Please check the system details number on your camera.
> If it does not start with S07, please upgrade it with the firmware at the link, https://drive.google.com/file/d/1qCkdqhuGttK6mXdMUfQ1DTt9oHWNqBgc/view?usp=share_link.
>  
> ![image](https://user-images.githubusercontent.com/48876201/234319235-ecbaa929-149f-406c-9521-3442a52de567.png)
> 
> Let us know if there are any further concerns.
> Thanks for your understanding.
> 
> Best regards,
> Reolink Support Team -Olivia

After this exchange, and when I upgraded to this firmware, I asked:
> Thank you, I have upgraded it. Do you happen to any beta firmware that I can set a fixed frame rate and fixed bit rate?

They then sent me the firmware I shared. They are looking into the fixed frame rate option right now.

I am _ASSUMING_ the firmware I shared is under the same restrictions as the first one they sent me a link to.

**Reply by AT0myks** [reply]
Thanks for the details. I've updated the readme with your information and added the firmware you linked.

**Reply by clsferguson** [reply]
> Thanks for the details. I've updated the readme with your information and added the firmware you linked.

Yeah, happy to help. Here is the [link](https://support-d.reolink.com/attachments/token/4SXDbC9daidjZw02rDj36QeLb/?name=IPC_51316M.757_21121503.RLC-410W.ov4689.4MP.WIFI1021.REOLINK.pak) for the firmware they sent me in an email, I don't know how long it will be valid for.

They sent me a follow up email too.

> Dear Cory,
>  
> This is Ian from the Reolink Senior Support Team. 
>  
> After confirming with the engineers, we have come to the conclusion that the current version supports a fixed frame rate as default, meaning there is no option available to change or set up the rate. Do you agree to use the camera this way?
>  
> Thanks for your cooperation.
> Best Regards,
> Reolink Senior Support Team - Ian

I think this is incorrect information though.

FFPROBE on the camera outputs this:
```
FFPROBE OUTPUT
Stream 0:
Return Code: 0

Video:

Codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Resolution: 2560x1440
FPS: 355/12

Audio:

Codec: AAC (Advanced Audio Coding)
```
Sometimes FPS comes back as unknown.

**Reply by AT0myks** [reply]
That's what I consider an "official" link since the file is hosted by Reolink. I prioritize these links so I added it too, it's always better to download straight from the source. I've never seen a dead one from them yet.

The `FPS` in the ffprobe output is a fraction, so right here you get 29.58333333333333..., meaning your stream is set to 30 FPS. And I'm not sure there are a lot of IP cameras that can do variable frame rate anyway.
My guess would be that sometimes ffprobe doesn't receive enough data about the stream so it reports unknown FPS. You might want to try using ffmpeg with options like `analyzeduration` and `probesize`.
